### PR TITLE
fix: test cases to show that fixer does not work with gjs and gts files

### PR DESCRIPTION
**Pull request by gabrielcsapo**
# Summary

Adds the fix to the issue showcased in https://github.com/ember-template-lint/ember-template-lint/pull/2696

### CODE REVIEW

File: lib/linter.js [hunk 1]
```diff
@@ -230,17 +230,23 @@ export default class Linter {
...
+          shouldFix: true,
+          filePath: options.filePath,
+          rawSource: templateInfo.template,
+          fileConfig,
+        });
+
+        let visitor = await rule.getVisitor();
+        let { code } = transform(templateInfo.template, () => visitor);
+
+        currentSource = currentSource.replace(templateInfo.template, code);
```
Comment (rwjblue) [inline lib/linter.js:248]: Can you add a test case for:

```js
// this one shouldn't be touched by a `--fix`
const bar = <template>{{! <button></button> }}</template>;
// this should get `--fix`'ed
const foo = <template><button></button></template>;
```

I think the current `string.replace` system will replace both `bar` and `foo` there. I believe that we need to manually splice the string contents, something like:

```js
let currentSource = `
const bar = <template>{{! <button></button> }}</template>;
const foo = <template><button></button></template>;`
// -> '\nconst bar = <template>{{! <button></button> }}</template>;\nconst foo = <template><button></button></template>;'

currentSource = currentSource.slice(0, 82) + `<button type="submit"></button>` + currentSource.slice(99)
// -> '\nconst bar = <template>{{! <button></button> }}</template>;\nconst foo = <template><button type="submit"></button></template>;'
```

File: test/unit/rules/require-button-type-test.js [hunk 2]
```diff
@@ -142,18 +140,19 @@ generateRuleTests({
       fixedTemplate: '<form><button type="submit"></button></form>',
     },
     {
-      template: '/** silly example <button> usage */ <button><button>',
+      template: '<template>/** silly example <button/> usage */ <button></button></template>',
```
Comment (rwjblue) [inline test/unit/rules/require-button-type-test.js:143]:
```suggestion
      template: '<template>/** silly example <button> usage */ <button></button></template>',
```

I think the `button` in the comment probably shouldn't self-close

File: test/unit/rules/require-button-type-test.js [hunk 3]
```diff
@@ -142,18 +140,19 @@ generateRuleTests({
...
       meta: {
         filePath: 'layout.gjs',
       },
-      fixedTemplate: '/** silly example <button> usage */ <button type="button"></button>',
+      fixedTemplate:
+        '<template>/** silly example <button type="button" /> usage */ <button type="button"></button></template>',
```
Comment (rwjblue) [inline test/unit/rules/require-button-type-test.js:148]: This is proving my point in the other comment. We can't use `currentSource.replace`. The `button` in the comment should never be touched.

**Review by rwjblue — CHANGES_REQUESTED** [review]

File: test/unit/rules/require-button-type-test.js (hunk 2 above)
Comment (gabrielcsapo) [reply test/unit/rules/require-button-type-test.js:143]: I rebid on your branch upstream, good call.

**Pull request merged**